boa/src/syntax/lexer/mod.rs

```diff
@@ -399,16 +399,18 @@ impl<'a> Lexer<'a> {
                                     }
                                 }
                             }
+                            dbg!(&gone_decimal);
                             if gone_decimal {
-                                f64::from_str(&buf).map_err(|_e| LexerError::new("Could not convert value to f64"))?
+                                NumericLiteral::Number(f64::from_str(&buf).map_err(|_e| LexerError::new("Could not convert value to f64"))?)
                             } else if buf.is_empty() {
-                                0.0
+                                NumericLiteral::Integer(0)
                             } else {
-                                (u64::from_str_radix(&buf, 8).map_err(|_e| LexerError::new("Could not convert value to u64"))?) as f64
+                                NumericLiteral::Integer(
+                                u64::from_str_radix(&buf, 8).map_err(|_e| {dbg!(&_e); LexerError::new("Could not convert value to u64")})?)
```

Comment:
The `dbg!()` call should be removed before merging.
```suggestion
                                u64::from_str_radix(&buf, 8).map_err(|_e| LexerError::new("Could not convert value to u64"))?)
```